src/combinator.rs

```diff
@@ -1618,6 +1615,61 @@ where
     go_extra!(C);
 }
 
+/// See [`IterParser::collect_exactly`]
+pub struct CollectExactly<A, O, C, const N: usize> {
```

Comment:
I wonder whether the `N` should become an associated type of `ContainerExactly` instead of a type parameter? Then we don't need to have this `N` hanging around in the type signatures when really it's already implied by the collection type

Reply:
We'd have to remove the impl for `()`, though that should already be covered by `repeated().exactly()`

Reply:
I think it's worth it to do that, personally